packages/next/src/server-build.ts

```diff
@@ -1128,11 +1128,11 @@ export async function serverBuild({
             );
           });
         }
+        let outputName = path.posix.join(entryDirectory, pageNoExt);
 
-        let outputName = normalizeIndexOutput(
-          path.posix.join(entryDirectory, pageNoExt),
-          true
-        );
+        if (!(group.isAppRouter || group.isAppRouteHandler)) {
```

Comment:
maybe a skill issue 😆  but I usually have to think a bit extra on these conditions

```suggestion
        if (!group.isAppRouter && !group.isAppRouteHandler) {
```